frontends/p4/frontend.h

```diff
@@ -60,6 +60,18 @@ class FrontEndPolicy : public RemoveUnusedPolicy {
         return options.optimizationLevel > 0;
     }
 
+    /// Indicates whether control plane API generation is enabled.
+    /// @returns default to false unless a command line option was
+    /// given explicitly enabling control plane API generation.
+    virtual bool controlPlaneAPIGenEnabled(const CompilerOptions &options) const {
```

Comment:
This function should probably be part of the Options class, which owns these members. I believe CompilerOptions? You could also make this virtual since Options are usually target-specific.

Reply:
Although I also can see it make sense as part of the FrontEndPolicy.

Reply:
I moved method controlPlaneAPIGenEnabled into class CompilerOptions in commits 47 & 48.